### PR TITLE
Improve OpenGL debugging

### PR DESCRIPTION
**Pull request by Gustl22**

### CODE REVIEW

**Comment by Gustl22** [comment]
Only wanted to know if you're busy @devemux :wink:

**Comment by devemux86** [comment]
Being around, looking for free time! :slightly_smiling_face:

File: vtm/src/org/oscim/renderer/GLUtils.java [hunk 1]
```diff
@@ -107,26 +109,89 @@ public static int loadTexture(byte[] pixel, int width, int height, int format,
...
-        while ((error = gl.getError()) != 0) { // GL20.NO_ERROR) {
-            log.error(op + ": glError " + error);
+        int error; // GL.NO_ERROR
+        List<Integer> errors = new ArrayList<>();
+        while ((error = gl.getError()) != GL.NO_ERROR) {
+            log.error(op + ": \tglError " + getGlErrorString(error) + " (" + error + ")");
+            errors.add(error);
             // throw new RuntimeException(op + ": glError " + error);
         }
+        return errors;
```
Comment (devemux86) [inline vtm/src/org/oscim/renderer/GLUtils.java]: Better use heavier implementations with list creation when needed.
And keep the simpler one for all other frequent cases.

**Comment by devemux86** [comment]
Thanks, merged via 4b418b9bb910c6a7dc751c30c9b242d0eb6f1fc7 with the mentioned improvement.

File: vtm/src/org/oscim/renderer/GLUtils.java (hunk 1 above)
Comment (Gustl22) [reply vtm/src/org/oscim/renderer/GLUtils.java]: Good idea :+1:

**Comment by Gustl22** [comment]
I think limit to `GL.OUT_OF_MEMORY` isn't a good solution. There shouldn't be extra methods for each single error.
Why not overload `checkGlError(String op, int error)` and return true, if `error` occurs?

**Comment by devemux86** [comment]
Feel free to PR another implementation. :slightly_smiling_face: 
But performance should always be in our mind.

**Pull request closed without merging**